This dependency was introduced in
https://github.com/tensorflow/tensorflow/commit/c211472000ff57bac7fcec9b0465cf73b37bf135
> This is in preparation to open-source TF's TFRT backend.
> TFRT generates code using libclang python bindings as part of the build.

Hence it is not currently used and as it is not (easily) available for all architectures
simply remove it.

Author: Alexander Grund (TU Dresden)

diff --git a/tensorflow/tools/pip_package/setup.py b/tensorflow/tools/pip_package/setup.py
index a9ef0ee00b2..6740e2fe3a7 100644
--- a/tensorflow/tools/pip_package/setup.py
+++ b/tensorflow/tools/pip_package/setup.py
@@ -82,7 +82,6 @@ REQUIRED_PACKAGES = [
     # Install other dependencies
     'absl-py >= 0.4.0',
     'astunparse >= 1.6.0',
-    'libclang >= 9.0.1',
     'flatbuffers >= 1.12, < 3.0', # capped as jax 0.1.71 needs < 3.0
     'google_pasta >= 0.1.1',
     'h5py >= 2.9.0', # capped since 3.3.0 lacks py3.6
